feat: hosted ui for flutter (#2864)
aws-amplify/docs: docs/lib/auth/fragments/flutter/signin_web_ui/30_signin.md

> Sweet! You're now ready to launch sign in with web UI. 

Reviewer: 🍬 
```suggestion
Sweet! You're now ready to launch sign-in with web UI. 
```

Author: Taken directly from android docs.

```dart
try {
  SignInResult res = await Amplify.Auth.signInWithWebUI();
} on AuthError catch (e) {
  print(e);
}
```

> You can also specify a provider with the `provider` attribute.

Reviewer: ```suggestion
You can also specify a provider with the `provider` attribute:
```

```dart
try {
  SignInResult res = await Amplify.Auth.signInWithWebUI(provider:  AuthProvider.google);
} on AuthError catch (e) {
  print(e);
}
```

> Amplify-Flutter currently supports the following social login providers: 

Reviewer: ```suggestion
Amplify Flutter currently supports the following social sign-in providers: 
```
